testutils/testutils.go

```diff
@@ -131,29 +134,117 @@ func MockSystemState() engine.SystemState {
 }
 
-func MockQueryStateResponse() jsonrpc.ResponseQueryState {
-	bitcoinState := pack.NewStruct(
-		"pubKey", pack.String("Akwn5WEMcB2Ff_E0ZOoVks9uZRvG_eFD99AysymOc5fm"),
-		"gasCap", pack.U64(2),
-		"gasLimit", pack.U64(3),
-	)
+func MockQueryBlockStateResponse() jsonrpc.ResponseQueryBlockState {
+
+	bitcoinState, err := pack.Encode(MockEngineState()["BTC"])
+	if err != nil {
+		panic(fmt.Sprintf("failed to encode bitcoinState:  %v", err))
+	}
+	bitcoinStateS := bitcoinState.(pack.Struct)
+
+	terraState, err := pack.Encode(MockEngineState()["LUNA"])
+	if err != nil {
+		panic(fmt.Sprintf("failed to encode terraState:  %v", err))
+	}
+	terraStateS := terraState.(pack.Struct)
 
-	partialState := pack.NewStruct(
-		"pubKey", pack.String("Akwn5WEMcB2Ff_E0ZOoVks9uZRvG_eFD99AysymOc5fm"),
-	)
 	systemStateP, err := pack.Encode(MockSystemState())
 	if err != nil {
-		panic("Shouldn't fail")
-	}
-	return jsonrpc.ResponseQueryState{
-		State: map[pack.String]pack.Struct{
-			pack.String("System"):                systemStateP.(pack.Struct),
-			pack.String(string(multichain.BTC)):  bitcoinState,
-			pack.String(string(multichain.BCH)):  bitcoinState,
-			pack.String(string(multichain.ZEC)):  bitcoinState,
-			pack.String(string(multichain.LUNA)): partialState,
+		panic(fmt.Sprintf("failed to encode system state:  %v", err))
+	}
+
+	var state pack.Typed
+	state = append(state, pack.NewStructField("System", systemStateP))
+	state = append(state, pack.NewStructField(string(multichain.BTC), bitcoinStateS))
+	state = append(state, pack.NewStructField(string(multichain.BCH), bitcoinStateS))
+	state = append(state, pack.NewStructField(string(multichain.ZEC), bitcoinStateS))
+	state = append(state, pack.NewStructField(string(multichain.LUNA), terraStateS))
+
+	return jsonrpc.ResponseQueryBlockState{
+		State: state,
+	}
+}
+
+func MockEngineState() map[string]engine.XState {
+	// systemStateP, err := json.Marshal(MockSystemState())
+	// if err != nil {
+	// 	panic(fmt.Sprintf("Failed to encode state %v", err))
+	// }
+	// var systemX engine.XState
+	// err = json.Unmarshal(systemStateP, &systemX)
+	// if err != nil {
+	// 	panic(fmt.Sprintf("Failed to decode state %v", err))
+	// }
+	//
+	pkBytes, err := base64.RawURLEncoding.DecodeString("Akwn5WEMcB2Ff_E0ZOoVks9uZRvG_eFD99AysymOc5fm")
+	if err != nil {
+		panic(fmt.Sprintf("Failed to decode pubk %v", err))
+	}
+
+	bitcoinShardState := engine.XStateShardUTXO{
+		Outpoint: utxo.Outpoint{
+			Hash:  []byte{0},
+			Index: 0,
+		},
+		Value:        pack.NewU256FromU8(0),
+		PubKeyScript: pkBytes,
+	}
+	bitcoinShardStateP, err := pack.Encode(bitcoinShardState)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to encode utxo shard %v", err))
+	}
+
+	accountShardState := engine.XStateShardAccount{
+		Nonce:   pack.NewU256FromU64(0),
+		Gnonces: []engine.XStateShardGnonce{},
+	}
+	accountShardStateP, err := pack.Encode(accountShardState)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to encode account shard %v", err))
+	}
```

Comment:
```suggestion
	if err != nil {
		panic(fmt.Sprintf("failed to decode pub key: %v", err))
	}

	bitcoinShardState := engine.XStateShardUTXO{
		Outpoint: utxo.Outpoint{
			Hash:  []byte{0},
			Index: 0,
		},
		Value:        pack.NewU256FromU8(0),
		PubKeyScript: pkBytes,
	}
	bitcoinShardStateP, err := pack.Encode(bitcoinShardState)
	if err != nil {
		panic(fmt.Sprintf("failed to encode utxo shard: %v", err))
	}

	accountShardState := engine.XStateShardAccount{
		Nonce:   pack.NewU256FromU64(0),
		Gnonces: []engine.XStateShardGnonce{},
	}
	accountShardStateP, err := pack.Encode(accountShardState)
	if err != nil {
		panic(fmt.Sprintf("failed to encode account shard: %v", err))
	}
```

Just added some colons to help separate the error messages.